src/Compiler/Facilities/prim-lexing.fs

```diff
@@ -108,6 +109,46 @@ type StringText(str: string) =
         member _.CopyTo(sourceIndex, destination, destinationIndex, count) =
             str.CopyTo(sourceIndex, destination, destinationIndex, count)
 
+        member this.GetSubTextFromRange(range) =
+            let totalAmountOfLines = getLines.Value.Length
+
+            if
+                range.StartLine = 0
+                && range.StartColumn = 0
+                && range.EndLine = 0
+                && range.EndColumn = 0
+            then
+                String.Empty
+            elif
+                range.StartLine < 1
+                || (range.StartLine - 1) > totalAmountOfLines
+                || range.EndLine < 1
+                || (range.EndLine - 1) > totalAmountOfLines
+            then
+                invalidArg (nameof range) "The range is outside the file boundaries"
+            else
+                let sourceText = this :> ISourceText
+                let startLine = range.StartLine - 1
+                let line = sourceText.GetLineString startLine
+
+                if range.StartLine = range.EndLine then
+                    let length = range.EndColumn - range.StartColumn
+                    line.Substring(range.StartColumn, length)
+                else
+                    let firstLineContent = line.Substring(range.StartColumn)
+
+                    let capacity =
+                        [ (range.StartLine - 1) .. (range.EndLine - 1) ]
+                        |> List.sumBy (fun lineIdx -> (sourceText.GetLineString lineIdx).Length)
+
+                    let sb = System.Text.StringBuilder(capacity).AppendLine(firstLineContent)
```

Comment:
This is at the end of the day quite possibly more expensive than having the string builder start with the default capacity and resize on the go.